### PR TITLE
[bug_5702] Added external file with personal users data. Added pictures....

### PR DESCRIPTION
**Pull request by MarcinWyt**
Proszę o review.

https://gist.github.com/MarcinWyt/6f7a0e8a531c54b2467a

https://bugzilla.stxnext.pl/show_bug.cgi?id=5702

### CODE REVIEW

File: src/presence_analyzer/config.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+)
+USERS_XML_LOCAL_FILE = os.path.join(
+    USERS_XML_LOCAL_DIR, 'users.xml'
+)
+USERS_XML_TEST_FILE = os.path.join(
+    USERS_XML_LOCAL_DIR, 'users_test.xml'
+)
+
+# USERS_XML_REMOTE = "http://sargo.bolt.stxnext.pl/users.xml"
+# USERS_XML_LOCAL = "${buildout:directory}/runtime/data/users.xml"
```
Comment (dpolcyn) [inline src/presence_analyzer/config.py]: You don't need this.

File: src/presence_analyzer/templates/presence_weekday.html [hunk 2]
```diff
@@ -41,4 +57,4 @@
 
 <%block name="block_title">
     Presence mean time by weekday
-</%block>
+</%block>
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_weekday.html]: Enter after this line.

File: src/presence_analyzer/views.py [hunk 3]
```diff
@@ -66,6 +68,34 @@ def users_view():
...
+@jsonify
+def users_view_data():
+    """
+    Users listing for dropdown.
+    """
+    # pylint: disable=no-member
+    # with open(app.config['USERS_XML_REMOTE_FILE'], 'r') as f_xml:
+    #     tree = etree.parse(f_xml)
+    tree = etree.parse(app.config['USERS_XML_LOCAL_FILE'])
+    url_prefix = \
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Use format.

File: src/presence_analyzer/tests.py [hunk 4]
```diff
@@ -57,12 +63,46 @@ def test_api_users(self):
...
+        ]
+        self.assertListEqual(json.loads(resp.data), expected_data)
+
     def test_mean_time_weekday_view(self):
         """
         Test mean time weekday view.
         """
         resp = self.client.get('/api/v1/mean_time_weekday/10000')
-        self.assertEqual(resp.status_code, 404)
+        # self.assertEqual(resp.status_code, 404)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You don't need this.

File: src/presence_analyzer/tests.py [hunk 5]
```diff
@@ -85,7 +125,8 @@ def test_presence_weekday_view(self):
         Test presence weekday view.
         """
         resp = self.client.get('/api/v1/presence_weekday/10000')
-        self.assertEqual(resp.status_code, 404)
+        # self.assertEqual(resp.status_code, 404)
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: You don't need this.

File: src/presence_analyzer/views.py [hunk 6]
```diff
@@ -116,7 +148,8 @@ def presence_start_end_view(user_id):
     data = get_data()
     if user_id not in data:
         log.debug('User %s not found!', user_id)
-        abort(404)
+        # abort(404)
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Don't leave commented out code in the PR.

File: src/presence_analyzer/views.py [hunk 7]
```diff
@@ -66,6 +68,34 @@ def users_view():
...
+        '://' + \
+        tree.find('server').find('host').text + \
+        ':' + \
+        tree.find('server').find('port').text
+
+    result = [
+        {
+            'user_id': person.get('id'),
+            'name': person.findtext('name'),
+            'avatar': url_prefix + person.findtext('avatar'),
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Use format here as well.

File: src/presence_analyzer/views.py [hunk 8]
```diff
@@ -66,6 +68,34 @@ def users_view():
...
 
+@app.route('/api/v1/users_data', methods=['GET'])
+@jsonify
+def users_view_data():
+    """
+    Users listing for dropdown.
+    """
+    # pylint: disable=no-member
+    # with open(app.config['USERS_XML_REMOTE_FILE'], 'r') as f_xml:
+    #     tree = etree.parse(f_xml)
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: You don't need this.

File: src/presence_analyzer/views.py [hunk 9]
```diff
@@ -95,7 +126,8 @@ def presence_weekday_view(user_id):
     data = get_data()
     if user_id not in data:
         log.debug('User %s not found!', user_id)
-        abort(404)
+        # abort(404)
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: You don't need this.

File: src/presence_analyzer/views.py (hunk 6 above)
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: You don't need this.

File: src/presence_analyzer/views.py [hunk 10]
```diff
@@ -3,11 +3,13 @@
 Defines views.
 """
 
-# p y lint: disable=import-error, no-name-in-module
+# pylint: disable=import-error, no-name-in-module
 import calendar
 from flask import redirect, abort
 from flask.ext.mako import render_template, exceptions
 
+from lxml import etree
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: Please sort the imports according to https://www.python.org/dev/peps/pep-0008/#imports

File: setup.py [hunk 11]
```diff
@@ -28,10 +28,12 @@ def read(*rnames):
         'setuptools',
         'Flask',
         'Flask-Mako',
+        'lxml'
     ],
     entry_points="""
     [console_scripts]
     flask-ctl = presence_analyzer.script:run
+    cron = presence_analyzer.config:get_user_xml
```
Comment (dpolcyn) [inline setup.py]: ```cron```? I suggest ```update_users_xml```

File: src/presence_analyzer/users_cron.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    USERS_XML_LOCAL_FILE
+)
+
+
+def get_users_xml():
+    """
+    Download actual users data XML.
+    """
+    f_remote = urllib2.urlopen(USERS_XML_REMOTE_FILE)
+    f_local = open(USERS_XML_LOCAL_FILE, "w")
```
Comment (mjelonek92) [inline src/presence_analyzer/users_cron.py]: Use ```with``` statement.

File: src/presence_analyzer/users_cron.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    data = f_remote.read()
+    f_local.write(data)
+    f_local.close()
+    f_remote.close()
+    return data
```
Comment (mjelonek92) [inline src/presence_analyzer/users_cron.py]: Why return it?

File: src/presence_analyzer/utils.py [hunk 14]
```diff
@@ -103,7 +103,7 @@ def mean(items):
     """
     return float(sum(items)) / len(items) if len(items) > 0 else 0
 
-
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: Expected 2 empty lines.

File: src/presence_analyzer/utils.py [hunk 15]
```diff
@@ -103,7 +103,7 @@ def mean(items):
...
+#pylint: disable=invalid-name
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: ```# pylint: disable=invalid-name```

File: src/presence_analyzer/users_cron.py [hunk 16]
```diff
@@ -0,0 +1,28 @@
+# -*- encoding: utf-8 -*-
+"""
+Python script for cron.
+"""
+
+import urllib2
+#pylint: disable=relative-import
```
Comment (dpolcyn) [inline src/presence_analyzer/users_cron.py]: ```# pylint: disable=relative-import```

**Comment by dpolcyn** [comment]
Run tests.

**Comment by dpolcyn** [comment]
[tests.py-L12](https://github.com/stxnext-kindergarten/presence-analyzer-mwytrzyszczewski/pull/7/files#diff-06804ee992f7c6b264fc1882858d9b78L12) you don't need this import.

**Comment by dpolcyn** [comment]
I can't see any data. Please fix it. ```GET http://127.0.0.1:8080/api/v1/users_data 500 (INTERNAL SERVER ERROR)```

File: src/presence_analyzer/tests.py [hunk 17]
```diff
@@ -57,12 +62,45 @@ def test_api_users(self):
...
+                u'user_id': u'176',
+                u'name': u'Adrian K.',
+                u'avatar': u'https://intranet.stxnext.pl:443'
+                           u'/api/images/users/176',
+            },
+            {
+                u'user_id': u'170',
+                u'name': u'Agata J.',
+                u'avatar': u'https://intranet.stxnext.pl:443'
+                           u'/api/images/users/170',
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: Unicode.

File: src/presence_analyzer/users_cron.py [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+
+def get_users_xml():
+    """
+    Download actual users data XML.
+    """
+    f_remote = urllib2.urlopen(USERS_XML_REMOTE_FILE)
+    data = f_remote.read()
+    with open(USERS_XML_LOCAL_FILE, "w") as f_local:
+        f_local.write(data)
+        f_local.close()
```
Comment (mjelonek92) [inline src/presence_analyzer/users_cron.py]: No need to close when you use ```with```.

File: src/presence_analyzer/views.py [hunk 19]
```diff
@@ -3,10 +3,11 @@
 Defines views.
 """
 
-# p y lint: disable=import-error, no-name-in-module
+# pylint: disable=import-error, no-name-in-module
 import calendar
 from flask import redirect, abort
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: ```abort``` you don't need this anymore.

File: src/presence_analyzer/templates/presence_start_end.html [hunk 20]
```diff
@@ -7,48 +7,64 @@
...
                     });
                     dropdown.show();
                     loading.hide();
                 });
                 $('#user_id').change(function() {
                     var selected_user = $("#user_id").val(),
-                        chart_div = $('#chart_div');
+                        chart_div = $('#chart_div'),
+                        avatar_url = $('option:selected', this).attr('avatar');
                     if(selected_user) {
```
Comment (dpolcyn) [inline src/presence_analyzer/templates/presence_start_end.html]: ```if (selected_user) {```

**Comment by dpolcyn** [comment]
```Presence by weekday``` tab, I see tittle ```Presence mean time by weekday```. Please fix it.

File: src/presence_analyzer/tests.py [hunk 21]
```diff
@@ -55,14 +61,47 @@ def test_api_users(self):
...
+        ]
+        self.assertListEqual(json.loads(resp.data), expected_data)
 
     def test_mean_time_weekday_view(self):
         """
         Test mean time weekday view.
         """
         resp = self.client.get('/api/v1/mean_time_weekday/10000')
-        self.assertEqual(resp.status_code, 404)
+        self.assertEqual(resp.data, '"no_data"')
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```self.assertEqual(json.loads(resp.data), 'no_data')```

File: src/presence_analyzer/tests.py [hunk 22]
```diff
@@ -85,7 +124,7 @@ def test_presence_weekday_view(self):
         Test presence weekday view.
         """
         resp = self.client.get('/api/v1/presence_weekday/10000')
-        self.assertEqual(resp.status_code, 404)
+        self.assertEqual(resp.data, '"no_data"')
```
Comment (dpolcyn) [inline src/presence_analyzer/tests.py]: ```self.assertEqual(resp.data, 'no_data')```

**Comment by dpolcyn** [comment]
Please hold tab order: ```presence by weekday, presence mean time, presence start end```

File: src/presence_analyzer/views.py [hunk 23]
```diff
@@ -75,7 +101,7 @@ def mean_time_weekday_view(user_id):
     data = get_data()
     if user_id not in data:
         log.debug('User %s not found!', user_id)
-        abort(404)
+        return "no_data"
```
Comment (dpolcyn) [inline src/presence_analyzer/views.py]: ```return 'no_data'```

**Pull request merged**